### ISSUE TITLE
Guild incorrectly importing flags from other modules

### ISSUE THREAD

**Issue by jethrokuan** [guildai/guildai#119]
On a `guild run op` command, flags that do not belong to my argument parser are being imported:

Example argument parser:

```
(miniconda3-4.3.30) jethro@beast-CLeAR:~/visuo-tactile-snn$ python train.py --help
usage: Trainer. [-h] --epochs EPOCHS --data_dir DATA_DIR --checkpoint_dir
                CHECKPOINT_DIR --tsample TSAMPLE --tsr_stop TSR_STOP --sc_true
                SC_TRUE --sc_false SC_FALSE --lr LR --sample_file SAMPLE_FILE
                --hidden_size HIDDEN_SIZEoptional arguments:
  -h, --help            show this help message and exit
  --epochs EPOCHS       Number of epochs.
  --data_dir DATA_DIR   Path to data.
  --checkpoint_dir CHECKPOINT_DIR
                        Path for saving checkpoints.
  --tsample TSAMPLE     tSample
  --tsr_stop TSR_STOP   Target Spike Region Stop
  --sc_true SC_TRUE     Spike Count True
  --sc_false SC_FALSE   Spike Count False
  --lr LR               Learning rate.
  --sample_file SAMPLE_FILE
                        Sample number to train from.
  --hidden_size HIDDEN_SIZE
                        Size of hidden layer.
```

Guild op configuration:

```
    baseline:
      description: Train baseline model
      main: train --checkpoint_dir .
      output-scalars: no
      flags-import: all
      flags-import-skip: [checkpoint_dir]
      flags:
        epochs:
          default: 1500
        lr:
          default: 0.01
        data_dir:
          default: data
```

Guild run command: 
`guild run vitac:baseline prepared-data=e3b0c29f data_dir=-- tsample=65 tsr_stop=65 sc_true=50 sc_false=3 hidden_size=80 sample_file=1 epochs=3000 lr=0.1 --label tas_repro_2 -by --gpus 2`

Run errors out, and `guild run info 1` shows:

```
flags:
  append: no
  bg: no
  breakpoint: null
  cell: no
  data_dir: --
  epochs: 3000
  err: null
  exit: no
  export: no
  hidden_size: 80
  instance: no
  isolated: no
  line: no
  list: no
  lr: 0.1
  no-display: no
  no-import-all: null
  no-raise-error: yes
  no-stderr: no
  no-stdout: no
  out: null
  params: null
  prepared-data: e3b0c29f
  proc: null
  sample_file: 1
  sc_false: 3
  sc_true: 50
  tsample: 65
  tsr_stop: 65
  yes: no
```

**Comment by gar1t** [comment]
I create an issue tracking doc for this:

https://github.com/guildai/issue-resolution/tree/master/119-guild-incorrectly-importing-flags-from-other-modules

This should is fixed in 0.7.0.rc5.

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0. If this is still an issue, please re-open and we'll sort it out. Thanks again for the report!
